### PR TITLE
UCT/CUDA: Support new gdrcopy API.

### PR DESCRIPTION
**Pull request by ssaulters**
## What
_Adding support for new versions of gdrcopy_

## Why ?
_Breaking changes in gdrcopy API_

@bureddy

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by hoopoepg** [comment]
ok to test

**Comment by shamisp** [comment]
@ssaulters Thanks for the patch! What organization do you represent ? I'm asking since we have to verify CLA for the contribution. Much appreciated.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12167/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 3 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/e21ab417f329270f328f7a2f337f758251c1c1bf/_main) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/edc11e64e7f4a5fb815138024a7baab0c7280e39/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/63e24e6e0b4574df2f0dcff87df83e3f546b8c11/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/3d0509e5ea4c1aba26c8de522865651621189ee3/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/bb02ae7c583c9ed41a1571e645c806fee647ab0a/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/e6cdb1896c6663e9aa30521aa805dfa78b02a4f4/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/895f0410b963bf6627e8f96ab78cc8cc85279508/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/4959dee873ca4e75bdee15cf3c8d335e29728d2d/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/a6279d35f32461ac62eca7a48fb493fafe069253/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/47fea2708e8a930a48480b98a7d4c423b257c806/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/1290b87ba14937a21b3117d79bd880d97f94b1a5/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/4b4c4eb1dbb2865eb8b013228357e0a4698fffc4/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/cb0fd4a0aae62ab30935cae05f20001d7abaee81/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/8714915a9b148cfcf1f5c8813275ba320a588819/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/78984037a741def02d67175df0bdca8e0117ac1a/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/07abfaf95bdae9f4922880b3c7a4f28222746301/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/eae1bde55317824263ddcfb8b096ab93beb277c1/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/ecb235ebad02439076005b5d4265aaf590e38d2f/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/5a8e31735e8b8fb5e9d929bcda087acb72c0e563/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/d06d5ec707395eea6d7275b8bb83fe20f4bdb6e1/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/377190c2a6488651ed4dc1f342a20787e957da9a/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/09a1dea005bfbbb78be6b6c25d40334ba8fa270c/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/8aa08187f770132bd6e699899b6a97a59d9d722d/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/d1f32a762bcf30477d6d7693b43c8a3de7fa9bf6/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/cfe6f4a93b594f4ec85a142fef7114ecc0a4044b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/325a0bcb21e8e00e09f20e9ca09f474b79378bba/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/d5d951c69983737b3ec1eae4c3d9f1772c4d788c/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/f598da1b991c5cfe92ce2c6dacaec7ce1b7d516f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f8487044609a230cb36e66ea3d629751/raw/370207d9757923cd1c57ab42a0776e2e341cea08/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by bureddy** [comment]
> @ssaulters Thanks for the patch! What organization do you represent ? I'm asking since we have to verify CLA for the contribution. Much appreciated.

@shamisp Scott is from NVIDIA

File: src/uct/cuda/gdr_copy/gdr_copy_ep.c [hunk 1]
```diff
@@ -12,6 +12,9 @@
 #include <ucs/sys/math.h>
 #include <ucs/type/class.h>
 
+#if (GDR_API_MAJOR_VERSION < 1 || (GDR_API_MAJOR_VERSION == 1 && GDR_API_MINOR_VERSION < 4))
+#define USE_OLD_GDRCOPY_API 1
```
Comment (yosefe) [inline src/uct/cuda/gdr_copy/gdr_copy_ep.c]: i think it's better to check for gdr_copy_to_mapping function presence by autoconf macro, in https://github.com/openucx/ucx/blob/master/config/m4/gdrcopy.m4

**Comment by shamisp** [comment]
Please add Nvidia copyright to the header of the files.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12240/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 12-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/1b64b4f0449c73c3b9569aadacebbc1863e8c058/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/653eadd8cdf4b980a7aa9cb7a81e6c9d5cbe510f/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/ccab154436ee38a30759787a37f91838383a5737/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/061911a21f97adf930609ff9dd77e1fdab83d664/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/226b3906d105f4be403be3e24043964b88a9205a/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/cfa30927d41db65e3626de9c176e5a36d9766695/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/1308086ef16acc6ff176afc1af6ecb8518ce575d/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/9b30b236441523fc0cb0086337fd2e57fa669a96/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/722193f63c7508327044adea202bb49bf9b02814/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/709d8fd9230f2182514fbe5bd509091eec5d4006/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/144c0f44aa99dcf64b35900b817c9286e1506fa9/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/64cdb81721592de274d4272c88913d250d9c3360/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/1021db0d7b1a1d43dc196a1219e09d800b1ad670/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/1ec483570801a9bef0c6be441c5d4e41d7522547/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/21820416d3cacedd665b7394b96217e18587f1fe/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/4d04924c3e62ea64d191f49b28f4d4e1ed7661ea/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/005e7fa21c27bdeea94de059b5cd79043ca7b3f2/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/7884753d4d1166924fc1739d27486bc7eadc7e38/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/897ad9c0f7046f7efdc80121f5f92beb8b3ca3a1/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/98e2d1a487df3ec579086c75c5b9b102dbb19a7d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/ca47593636c741f5f7e98ec010d8873fe0e2f9a0/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/92c8c2a2f40ec3e448415043f82afa82859d8634/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/422eea7779f649b2f1e1806f64532077fa9d9ac1/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/a3b9c07814fc0515390606dc39a338ebd0a741e0/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/4ad1d222896bdfc635018e002eede2a56806fd11/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/4c5ce98e548405293ab91c253062550ad9c7ec68/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/231608634c1a3fcefe263f36e39f225c812bf8ad/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/3e81a54b09022e2636279d62cc656b9ff68deff2/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/11f7657ca6a8e7a032d759e45e25169e/raw/035998223c1769f9bb06f16cbb815321fc720cc8/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: config/m4/gdrcopy.m4 [hunk 2]
```diff
@@ -38,20 +39,26 @@ AS_IF([test "x$with_gdrcopy" != "xno"],
...
+                [AC_MSG_NOTICE([gdrcopy using old API.])])])
+
         CFLAGS="$save_CFLAGS"
         CPPFLAGS="$save_CPPFLAGS"
         LDFLAGS="$save_LDFLAGS"
 
         AS_IF([test "x$gdrcopy_happy" = "xyes"],
             [
-                AC_SUBST(GDR_COPY_CPPFLAGS, "-I$ucx_check_gdrcopy_dir/include/ ")
+                AC_SUBST(GDR_COPY_CPPFLAGS, "-I$ucx_check_gdrcopy_dir/include/ $GDR_DEFINES")
```
Comment (bureddy) [inline config/m4/gdrcopy.m4]: @yosefe is this ok? or should we go with AC_CHECK_DECLS(gdr_copy_to_mappin,...)

**Comment by yosefe** [comment]
@ssaulters please rebase

File: config/m4/gdrcopy.m4 (hunk 2 above)
Comment (yosefe) [reply config/m4/gdrcopy.m4]: AC_CHECK_DECLS is better

File: config/m4/gdrcopy.m4 [hunk 3]
```diff
@@ -38,20 +39,26 @@ AS_IF([test "x$with_gdrcopy" != "xno"],
                             gdrcopy_happy="no"])
             ], [gdrcopy_happy="no"])
 
+        AS_IF([test "x$gdrcopy_happy" = "xyes"],
+            [AC_CHECK_LIB([gdrapi], [gdr_copy_to_mapping],
+                [GDR_DEFINES="-DUCT_CUDA_GDR_COPY_HAS_COPY_TO_MAPPING"],
```
Comment (yosefe) [inline config/m4/gdrcopy.m4]: GDR_DEFINES cppflags is not the way to do it
AC_CHECK_DECLS would produce HAVE_xx macro in config.h

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12305/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 16-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/b340f2e4d590409983efd3fdc49b8698bfc91984/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/dc01cda756510712411ccdb1927e422af033855b/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/44c005350c5d32aace24207d8bb69055a2455b7a/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/1fcab24bf64b75efb4768b37a3944a8c768ec7cc/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/3ad08fb68e14601e03ed5e57169da2e81fc8857f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/7ad94e9ae818ff2ebee5231c85795f44c0e34944/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/d3a49880ae28bd08e048baf9c4e7e08a11692333/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/d8b20a8606c9e6bb3a4ac721fb2105d0fefdede4/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/35e0b43f79f87ead23440c8eb3e8fe1ba7264961/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/36f982625f7f316d4820d176d5d68be881d2bb1e/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/bf82146e92cd5cb1b91bf4bb571cf921345c9963/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/9e708ce6d1460f397233d78818e74daedab54f52/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/9ae8fc0084bb9ee9f280642bf33594efebfd1fbd/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/b70a55a965f7097387c33ea75f706ecbdae83a9d/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/a67551cba87aff01fd2f1c08cfd50e68c939cec0/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/fd5a4e43e30ed639ac09cec55ecac3c97ba5171a/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/927a02a60b0f262e07a2f4abf809843f80803e6d/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/27c569148c63e8edaec40857ed51c717d39c2397/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/e3f97b74cbdbb870452fa0b41f745782eee09491/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/8ae772452d5cce0fd85aa359d0b6df38a45947b7/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/21ff8647699020a46fefda9ee5c6350dbb896ee7/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/e80e695c9ad3818b048cb2b525059d29eed32fd6/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/51c3f05ac0d6f3c85afecb5f998b4dfb739bcf73/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/bc12a92463cb560d664243461426a763be863d4d/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/8b7fa83406d3d60bbcd58f9e66cf9a23d3a9e5ec/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/7a8f1a59527381519a41c14ac3937d847cb5c402/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/a7f56d654a42b2e79d01555512d53eca7d79de70/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/f8184c1628a5043bc01d1f24cff39cc178221d9f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0b20a12fd7298220ea7fb56e32ca1384/raw/f59fc6ab0612a81fc521672f3f54d04c566f40d3/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@ssaulters please rebase

**Comment by yosefe** [comment]
@Akshay-Venkatesh i would suggest to add gdrcopy 2.0 compilation test to Azure pipelines (by PRing the necessary changes to UCX)

**Pull request merged**